src/main/scala/nl/knaw/dans/lib/logging/servlet/RequestLogFormatter.scala

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib.logging.servlet
+
+import javax.servlet.http.HttpServletRequest
+import org.scalatra.{ MultiParams, ScalatraBase }
+
+import scala.collection.JavaConverters._
+
+trait RequestLogFormatter {
+  this: ScalatraBase =>
+
+  /**
+   * Constructs the `String` to be logged by `ServletLogger` about this request.
+   * The resulted log line by default consists of:
+   *   - the request's method (GET, POST, etc.)
+   *   - the request's URL
+   *   - the IP address this request is coming from
+   *   - parameters sent with this request
+   *   - headers sent with this request
+   *
+   * Please note that no data in this log line is masked. If masking is required, please refer
+   * to the aggregated `MaskedLogFormatter` or the individual maskers in the `masked` package.
+   *
+   * @return the `String` to be logged
+   */
+  protected def formatRequestLog: String = {
+    val method = request.getMethod
+    val requestURL = request.getRequestURL.toString
+    val formattedHeaders = formatHeaders(getHeaderMap(request)).makeString
+    val formattedParams = formatParameters(multiParams).makeString
+    val formattedRemoteAddress = formatRemoteAddress(Option(request.getRemoteAddr).getOrElse(""))
+
+    // TODO perhaps more of https://github.com/scalatra/scalatra/blob/2.7.x/core/src/main/scala/org/scalatra/util/RequestLogging.scala#L70-L85
```

Comment:
Are we going to tackle this todo before the merge? Perhaps content_type would be nice to add

Reply:
tried it, but `content_type` seems already part of the headers that are listed, so it's probably not necessary.

@jo-pol what should we do with the TODO?

Reply:
Suggestion to remove the TODO

```suggestion
```